tuning/tune.py

```diff
@@ -820,6 +825,15 @@ def generate_constraints(
         )
     ]
 
+    # The shared memory constraint is best compatible with mmt.
+    # For other dispatch types, there are cases where the shared memory size
+    # may not align with the tile size.
+    if problem_size.dispatch_kind == DispatchKind.mmt:
```

Comment:
Is this specific to mmt? I'd expect this to be the same for all dispatch kinds. Could you check this against a conv and report back?